File: 2022/src/bin/day1/main.py
from pathlib import Path

INPUT_PATH = Path(__file__).parent.joinpath("input.txt")


def main():
    part_one()
    part_two()


def part_one():
    print(load_input()[0])


def part_two():
    print(sum(load_input()[0:3]))


def load_input() -> list[int]:
    sections = open(INPUT_PATH).read().split("\n\n")
    calories = map(lambda section: sum(map(int, section.splitlines())), sections)
    return sorted(calories, reverse=True)


if __name__ == '__main__':
    main()

File: 2022/src/bin/day1/main.rs
fn main() {
    part_one();
    part_two();
}

fn part_one() {
    let calories = input();
    dbg!(calories[0]);
}

fn part_two() {
    let calories = input();
    dbg!(calories[..3].into_iter().sum::<usize>());
}

fn input() -> Vec<usize> {
    let mut calories: Vec<usize> = include_str!("./input.txt")
        .split("\n\n")
        .map(|s| s.lines().map(|s| s.parse::<usize>().unwrap()).sum())
        .collect();
    calories.sort_unstable_by(|a, b| b.cmp(a)); // Sort descending
    calories
}
